tools/plugins/com.liferay.ide.project.ui/src/com/liferay/ide/project/ui/quickfix/LiferayDependencyQuickFix.java

```diff
@@ -0,0 +1,269 @@
+            servicesList = TargetPlatformUtil.getServicesList().getServiceList();
+
+            if( serviceWrapperList.contains( importName ) )
+            {
+                ServiceContainer bundle = TargetPlatformUtil.getServiceWrapperBundle( importName );
+                depWrapperCanFixed = true;
+                createDepProposal( context, proposals, bundle );
+            }
+
+            if( !depWrapperCanFixed )
+            {
+                if( servicesList.contains( importName ) )
+                {
+                    ServiceContainer bundle = TargetPlatformUtil.getServiceBundle( importName );
+                    createDepProposal( context, proposals, bundle );
+                }
+            }
+
+            if( TargetPlatformUtil.getThirdPartyBundleList( importName ) != null )
+            {
+                ServiceContainer bundle = TargetPlatformUtil.getThirdPartyBundleList( importName );
+                createDepProposal( context, proposals, bundle );
+            }
+        }
+        catch( Exception e )
+        {
+            ProjectCore.logError( "Project dependence got error", e );
+        }
+    }
+
+    private void undefinedType(
+        IInvocationContext context, IProblemLocation problem, Collection<IJavaCompletionProposal> proposals )
+    {
+        ASTNode selectedNode = problem.getCoveringNode( context.getASTRoot() );
+        String fullyQualifiedName = null;
+
+        if( selectedNode instanceof Name )
+        {
+            Name node = (Name) selectedNode;
+            fullyQualifiedName = node.getFullyQualifiedName();
+        }
+
+        List<String> serviceWrapperList;
+        List<String> servicesList;
+        boolean depWrapperCanFixed = false;
+
+        try
+        {
+            serviceWrapperList = TargetPlatformUtil.getServiceWrapperList().getServiceList();
+            servicesList = TargetPlatformUtil.getServicesList().getServiceList();
+
+            for( String wrapper : serviceWrapperList )
+            {
+                if( wrapper.endsWith( fullyQualifiedName ) )
+                {
+                    ServiceContainer bundle = TargetPlatformUtil.getServiceWrapperBundle( wrapper );
+                    createDepProposal( context, proposals, bundle );
+                }
+            }
+
+            if( !depWrapperCanFixed )
+            {
+                for( String service : servicesList )
+                {
+                    if( service.endsWith( fullyQualifiedName ) )
+                    {
+                        ServiceContainer bundle = TargetPlatformUtil.getServiceBundle( service );
+                        createDepProposal( context, proposals, bundle );
+                    }
+                }
+            }
+        }
+        catch( Exception e )
+        {
+            ProjectCore.logError( "Add module dependence got error", e );
+        }
+    }
+
+    private void createDepProposal(
```

Comment:
this method should return a new CUCorrectionProposal object that will be added to the proposals list